homeassistant/components/homekit_controller/connection.py

```diff
@@ -186,18 +199,7 @@ def async_set_unavailable(self):
 
         self.config_num = config_num
 
-        # For BLE, the Pairing instance relies on the entity map to map
-        # aid/iid to GATT characteristics. So push it to there as well.
-        self.pairing.pairing_data["accessories"] = self.accessories
-
-        self.async_load_platforms()
-
-        # Register and add new entities that are available
-        self.add_entities()
-
-        await self.async_update()
-
-        return True
+        self.hass.async_create_task(self.async_process_entity_map())
```

Comment:
Anything that uses `async_create_task` will be awaited before we move to the next phase of startup. So yeah, the web UI will always be up when this is done. Which leads me to wonder, why do anything in parallel to begin with ?

Reply:
(except for forward entry, which needs integration to be set up)

Reply:
The problem is that the platforms can't be loaded inside the initial `async_setup_entry`. So the code as it is on `dev` can do its first poll before they are. Thats what we are deferring here, i guess.

Reply:
So if i just take out this create task here i'll end up with an error that this aio task can't await itself, because its blocked on loading e.g. `homekit_controller.climate` which itself needs `homekit_controller`.

Reply:
Ok makes sense.